src/include/network/connection_handle.h

```diff
@@ -0,0 +1,212 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// connection_handle.h
+//
+// Identification: src/include/network/connection_handle.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <unordered_map>
+#include <event2/buffer.h>
+#include <event2/bufferevent.h>
+#include <event2/event.h>
+#include <event2/listener.h>
+
+#include <csignal>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <vector>
+
+#include <arpa/inet.h>
+#include <netinet/tcp.h>
+#include <sys/file.h>
+
+#include "common/exception.h"
+#include "common/logger.h"
+#include "type/types.h"
+
+#include "network/connection_handler_task.h"
+#include "protocol_handler.h"
+#include "marshal.h"
+#include "network_state.h"
+
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+
+namespace peloton{
+namespace network {
+
+// TODO(tianyu) This class is not refactored in full as rewriting the logic is not cost-effective. However, readability
+// improvement and other changes may become desirable in the future. Other than code clutter, responsibility assignment
+// is not well thought-out in this class. Abstracting out some type of socket wrapper would be nice.
+/**
+ * @brief A ConnectionHandle encapsulates all information about a client connection for its entire duration.
+ * One should not use the constructor to construct a new ConnectionHandle instance every time as it is expensive
+ * to allocate buffers. Instead, use the ConnectionHandleFactory.
+ *
+ * @see ConnectionHandleFactory
+ */
+class ConnectionHandle {
+public:
+
+
+
+  /**
+   * Update the existing event to listen to the passed flags
+   */
+  void UpdateEventFlags(short flags);
+
+  WriteState WritePackets();
+
+  std::string WriteBufferToString();
+
+  inline void HandleEvent(int, short) { state_machine_.Accept(Transition::WAKEUP, *this); }
+
+  // Exposed for testing
```

Comment:
Gross. People are going to use this when they probably shouldn't. I would make it protected and then have the tests be friend classes.